mas-foundation/src/main/java/com/ca/mas/foundation/MASDeviceMetadata.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2016 CA. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ *
+ */
+package com.ca.mas.foundation;
+import com.ca.mas.core.MobileSsoConfig;
+import com.ca.mas.core.store.StorageProvider;
+import com.ca.mas.foundation.notify.Callback;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class MASDeviceMetadata {
+
+    private static final String ENDPOINT_PATH = MASConfiguration.getCurrentConfiguration().getEndpointPath(MobileSsoConfig.DEVICE_METADATA_PATH);
+    private static final String MAG_IDENTIFIER = StorageProvider.getInstance().getTokenManager().getMagIdentifier();
+    private static final String HEADER_KEY = "mag-identifier";
+
+    private MASDeviceMetadata() {throw new IllegalStateException("Not allowed to instantiate");
+
+    }
+
+    public static void putAttribute(String attr, String value, final MASCallback<Void> callback) {
+
+        MASRequest request = null;
+        if (attr == null || !MASDevice.getCurrentDevice().isRegistered()) {
+            Callback.onError(callback, new Throwable());
+        }
+
+        try {
+            JSONObject data = new JSONObject();
+            data.put("name", attr);
+            data.put("value", value);
+
+            request = new MASRequest.MASRequestBuilder(new URI(ENDPOINT_PATH))
+                    .header(HEADER_KEY, MAG_IDENTIFIER)
```

Comment:
MAG_IDENTIFIER is not required, the SDK auto inject the mag-identifier for all request.